github/compliance/report.py

```diff
@@ -770,10 +790,23 @@ def sortable_result_str(result: gcm.ScanResult):
     if overwrite_repository:
         known_issues = _all_issues(overwrite_repository)
         issue_type = result_group_collection.issue_type
+
+        all_ctx_labels = set()
+
+        if not result_groups:
+            all_ctx_labels = result_group_collection.ctx_labels
```

Comment:
This is a potential error source resulting in unintended closed issues.

Given following example:
`_scanned_element_ctx_label` yields two ctx labels, `ctx/A` and `ctx/B`.
The passed `fallback` ctx labels (for no scan result case) is just `ctx/A`.

As the fallback is included, the issue filtering will yield the issue, thus "ignoring" `ctx/B`.
We check whether all `search_labels` are included in `issue_labels` (intersection, see [here](https://github.com/gardener/cc-utils/blob/bd5c0abac87b7ab3b2a609bd1f57176c33a23b8f/github/compliance/issue.py#L143)).

Reply:
We could consider dropping the "dynmaic" ctx label calculation and use the current "fallback" as single source of ctx labels, or tweak our issue filtering to ensure all ctx labels match (e.g. by defining a fixed ctx label prefix).